File: lab/workspace/work_java/src/b형대비특강/Day_0901/CodeName.java
package b형대비특강.Day_0901;

import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;

public class CodeName {
	
	static class Student{
		private String name;
		private int score;
		public Student(String name, int score) {
			super();
			this.name = name;
			this.score = score;
		}
		public String getName() {
			return name;
		}
		public void setName(String name) {
			this.name = name;
		}
		public int getScore() {
			return score;
		}
		public void setScore(int score) {
			this.score = score;
		}
		@Override
		public String toString() {
			return this.name + " " + this.score;
		}
	}
	
	static String[] temp;
	static int minScore = Integer.MAX_VALUE;
	
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
		Student[] students = new Student[5];
		for(int i=0; i<5; i++) {
			temp = br.readLine().split(" ");
			String name = temp[0];
			int score = Integer.parseInt(temp[1]);
			
			students[i] = new Student(name,score);
			students[i].setName(name);
			students[i].setScore(score);
			if(minScore > score) minScore = score;
		}
		for(Student s: students) {
			if(s.getScore() == minScore) {
				bw.write(s.toString());
				break;
			}
		}
		bw.flush();
		bw.close();
		
	}

}

File: First_Semester/lab/workspace/work_python/B형 대비 특강/Day_0901/객체/codename.py
# class Student():

#     def __init__(self, name='', score=0):
#         self.name = name
#         self.score = score
    
#     def __str__(self):
#         return '{0} {1}'.format(self.name, self.score)

# students = [
#     Student(),
#     Student(),
#     Student(),
#     Student(),
#     Student()
# ]
# min_score = float('inf')
# for i in range(5):
#     name, score = input().split()
#     students[i].name = name; students[i].score = int(score)
#     min_score = min(min_score, int(score))

# for student in students:
#     if student.score == min_score:
#         print(student)
#         break

dic = dict()
min_score = float('inf')
for _ in range(5):
    name, score = input().split()
    score = int(score)
    dic[score] = name
    if min_score > score: min_score = score
print(dic[min_score], min_score)